camel/models/nvidia_model.py

```diff
@@ -0,0 +1,188 @@
+    ) -> None:
+        if model_config_dict is None:
+            model_config_dict = NvidiaConfig().as_dict()
+        api_key = api_key or os.environ.get("NVIDIA_API_KEY")
+        url = url or os.environ.get(
+            "NVIDIA_API_BASE_URL", "https://integrate.api.nvidia.com/v1"
+        )
+        if token_counter is None:
+            # Use gpt-4 for token counting as it's compatible with most models
+            token_counter = OpenAITokenCounter(ModelType.GPT_4)
+        super().__init__(
+            model_type, model_config_dict, api_key, url, token_counter
+        )
+        self._client = OpenAI(
+            api_key=api_key,
+            base_url=url,
+            timeout=60,
+            max_retries=3,
+            default_headers={
+                "accept": "application/json",
+                "content-type": "application/json",
+            },
+        )
+
+    def check_model_config(self) -> None:
+        r"""Check if the model configuration contains any unexpected parameters.
+
+        Raises:
+            ValueError: If there are unexpected parameters in the configuration.
+        """
+        for param in self.model_config_dict:
+            if param not in NVIDIA_API_PARAMS:
+                raise ValueError(
+                    f"Unexpected parameter '{param}' for NVIDIA API."
+                )
+
+    def token_counter(self, text: str) -> int:
+        r"""Count the number of tokens in the given text.
+
+        Args:
+            text (str): The text to count tokens for.
+
+        Returns:
+            int: The number of tokens in the text.
+        """
+        return self._token_counter.count_tokens(text)
+
+    @api_keys_required("NVIDIA_API_KEY")
+    def run(
+        self,
+        messages: List[Dict[str, str]],
+        stream: bool = False,
+    ) -> Union[ChatCompletion, Stream[ChatCompletionChunk]]:
+        r"""Run the model with the given messages.
+
+        Args:
+            messages (List[Dict[str, str]]): List of messages for the conversation.
+            stream (bool, optional): Whether to stream the response.
+                (default: :obj:`False`)
+
+        Returns:
+            Union[ChatCompletion, Stream[ChatCompletionChunk]]: The model's response,
+            either as a complete message or a stream of chunks.
+        """
+        # Override stream parameter if specified in config
+        if 'stream' in self.model_config_dict:
+            stream = self.model_config_dict['stream']
+
+        # Prepare API parameters
+        api_params = {
+            "model": str(self.model_type),
+            "messages": messages,
+            "stream": stream,
+            "max_tokens": self.model_config_dict.get('max_tokens', 1024),
+            "temperature": self.model_config_dict.get('temperature', 0.5),
+            "top_p": self.model_config_dict.get('top_p', 1.0),
+            "frequency_penalty": self.model_config_dict.get('frequency_penalty', 0.0),
```

Comment:
there's some default value set in `model_config_dict `, could you refer to how other models implemented like below:

```
    def run(
        self,
        messages: List[OpenAIMessage],
    ) -> Union[ChatCompletion, Stream[ChatCompletionChunk]]:
        r"""Runs inference of Qwen chat completion.

        Args:
            messages (List[OpenAIMessage]): Message list with the chat history
                in OpenAI API format.

        Returns:
            Union[ChatCompletion, Stream[ChatCompletionChunk]]:
                `ChatCompletion` in the non-stream mode, or
                `Stream[ChatCompletionChunk]` in the stream mode.
        """
        response = self._client.chat.completions.create(
            messages=messages,
            model=self.model_type,
            **self.model_config_dict,
        )
        return response
```